### PR TITLE
lkl: mark it as broken on i686-linux

### PR DESCRIPTION
**Pull request by RaitoBezarius**
###### Description of changes

No one maintains i686 seriously enough, we will mark it as broken until miracle occurs.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
Commit should say that you are marking it broken.

File: pkgs/applications/virtualization/lkl/default.nix [hunk 1]
```diff
@@ -81,6 +81,7 @@ stdenv.mkDerivation rec {
     homepage    = "https://github.com/lkl/linux/";
     platforms   = platforms.linux; # Darwin probably works too but I haven't tested it
     license     = licenses.gpl2;
+    broken      = [ "i686-linux" ]; # No one really maintain it on i686 and it blocks all NixOS tests.
```
Comment (mweinelt) [inline pkgs/applications/virtualization/lkl/default.nix:84]:
```suggestion
    broken      = stdenv.isi686; # No one really maintain it on i686 and it blocks all NixOS tests.
```

File: nixos/doc/manual/release-notes/rl-2305.section.md [hunk 2]
```diff
@@ -341,6 +341,8 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 <!-- To avoid merge conflicts, consider adding your item at an arbitrary place in the list instead. -->
 
+- `i686-linux` NixOS tests requiring a system image (i.e. boot-related, etc.) are broken until [LKL](https://lkl.github.io/) is fixed on i686-linux or someone introduces a slowpath in `make-disk-image` for creating images without `cptofs` (fully in VM for example) which would only depend on QEMU then.
```
Comment (mweinelt) [inline nixos/doc/manual/release-notes/rl-2305.section.md:344]: Not sure breakages warrant release notes entries.

Comment (RaitoBezarius) [reply nixos/doc/manual/release-notes/rl-2305.section.md:344]: This is fairly big breakage for i686 users in NixOS, I'd say.

File: pkgs/applications/virtualization/lkl/default.nix (hunk 1 above)
Comment (RaitoBezarius) [reply pkgs/applications/virtualization/lkl/default.nix:84]: Done

**Comment by RaitoBezarius** [comment]
> Commit should say that you are marking it broken.

The commit message said it, I made the "title" more explicit.

**Comment by hesiod** [comment]
@RaitoBezarius Please hold off on merging this PR for the moment. I'm currently experimenting with a possible fix for the i686 build failure in #232902. I can't yet promise it's actually going to work, but I'd like to give it a try.

**Pull request closed without merging**